### PR TITLE
TablePanel: Refactor to functional component and move add ad hoc filter action to PanelContext 

### PR DESCRIPTION
**Pull request by torkelo**
When using Table panel via VizPanel from scenes you automatically got ad-hoc filters for most columns for a Prometheus query. 

* This makes it possible to disable ad-hoc filtering on a per-field basis (we where missing a registry item for filterable)
* It also moves the action to add new ad hoc filter to PanelContext , and from TablePanel to PanelStateWrapper
* Refactor TablePanel to functional component

### CODE REVIEW

File: public/app/plugins/panel/table/TablePanel.tsx [hunk 1]
```diff
@@ -1,164 +1,128 @@
...
+  let subData = subFrames;
+
+  if (!count || !hasFields) {
+    return <PanelDataErrorView panelId={id} fieldConfig={fieldConfig} data={data} />;
   }
 
-  onColumnResize = (fieldDisplayName: string, width: number) => {
-    const { fieldConfig } = this.props;
-    const { overrides } = fieldConfig;
+  let tableHeight = height;
```
Comment (codeincarnate) [inline public/app/plugins/panel/table/TablePanel.tsx:31]: Would be good to have this declared along with everything else

**Review by codeincarnate — COMMENTED** [review]
Code wise a few cleanups would be nice but otherwise looks good 😄

**Comment by torkelo** [comment]
> Code wise a few cleanups would be nice but otherwise looks good smile

@codeincarnate great! thanks for review! can do some final cleanups, but would be nice to get approved so I can merge it once that is done.

**Review by codeincarnate — APPROVED** [review]

**Comment by codeincarnate** [comment]
@torkelo done! 👍

**Review by ivanortegaalba — APPROVED** [review]
I like the reorg 👍

File: public/app/plugins/panel/table/TablePanel.tsx [hunk 2]
```diff
@@ -1,164 +1,128 @@
...
+  const frames = data.series;
+  const mainFrames = frames.filter((f) => f.meta?.custom?.parentRowIndex === undefined);
+  const subFrames = frames.filter((f) => f.meta?.custom?.parentRowIndex !== undefined);
+  const count = mainFrames?.length;
+  const hasFields = mainFrames[0]?.fields.length;
+  const currentIndex = getCurrentFrameIndex(mainFrames, options);
+  const main = mainFrames[currentIndex];
+  let subData = subFrames;
+
+  if (!count || !hasFields) {
```
[the remark this answers is not in the document]
Comment (torkelo) [reply public/app/plugins/panel/table/TablePanel.tsx:27]: @codeincarnate not sure what you had in mind here

**Pull request merged**